vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -293,181 +293,212 @@
 # ---------- ---------- ---------- ---------- ---------- ---------- ---------- ----------
 #                       Upgrade K8S version and boot-image                              |
 # ---------- ---------- ---------- ---------- ---------- ---------- ---------- ----------
-  - name: get cluster uid
-    shell: |
-            kubectl get clusters {{ CLUSTER_NAME }}  -n {{NAMESPACE}} -o json | jq '.metadata.uid' |   cut -f2 -d\"
-    environment:
-      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+  - name: Get cluster uid
+    community.kubernetes.k8s_info:
+      api_version: cluster.x-k8s.io/v1alpha3
+      kind: Cluster
+      name: {{ CLUSTER_NAME }}
+      namespace: {{ NAMESPACE }}
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
     register: CLSTR_UID
```

Comment:
What if you rename `CLSTR_UID` to `CLSTR_RESOURCE`. That way, it will not be confused with `CLUSTER_UID`

Reply:
Good point, will fix!